libs/cardano-ledger-core/src/Cardano/Ledger/Address.hs

```diff
@@ -791,13 +808,24 @@ decodeVariableLengthWord32 name buf = do
         res <- decode7BitVarLength name buf (\_ -> failDecoding name "too many bytes.") acc
         -- Only while decoding the last 7bits we check if there was too many
         -- bits supplied at the beginning.
-        unless (bufUnsafeIndex buf off0 .&. 0b01110000 == 0) $
+        unless (bufUnsafeIndex buf off0 .&. 0b11110000 == 0b10000000) $
           failDecoding name "More than 32bits was supplied"
         pure res
       {-# INLINE d7last #-}
   d7 (d7 (d7 (d7 d7last))) 0
 {-# INLINE decodeVariableLengthWord32 #-}
 
+-- | This decoder is here only with the purpose of preserving old buggy behavior. Should
+-- not be used for anything else.
```

Comment:
want to give it a scarier name?

Reply:
Nah, its alright. It is an internal function that is restricted to this module only. So the danger of anyone using it is almost non-existent.